modules/firefox/each-config.nix

```
@@ -0,0 +1,156 @@
+# manually applied args
+{
+  mkTarget,
+  name,
+  humanName,
+}:
+# module args
+{
+  lib,
+  pkgs,
+  ...
+}:
+mkTarget {
+  inherit name humanName;
+
+  extraOptions = {
+    profileNames = lib.mkOption {
+      description = "The ${humanName} profile names to apply styling on.";
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+    };
+
+    colorTheme.enable = lib.mkEnableOption "[Firefox Color](https://color.firefox.com/) on ${humanName}";
+
+    firefoxGnomeTheme.enable = lib.mkEnableOption "[Firefox GNOME theme](https://github.com/rafaelmardojai/firefox-gnome-theme) on ${humanName}";
+  };
+
+  configElements = [
+    (
+      { cfg }:
+      {
+        warnings =
+          lib.optional (cfg.profileNames == [ ])
+            ''stylix: ${name}: `config.stylix.targets.${name}.profileNames` is not set. Declare profile names with 'config.stylix.targets.${name}.profileNames = [ "<PROFILE_NAME>" ];'.'';
+      }
+    )
+    (
+      { cfg, fonts }:
+      {
+        programs.${name}.profiles = lib.genAttrs cfg.profileNames (_: {
+          settings = {
+            "font.name.monospace.x-western" = fonts.monospace.name;
+            "font.name.sans-serif.x-western" = fonts.sansSerif.name;
+            "font.name.serif.x-western" = fonts.serif.name;
+          };
+        });
+      }
+    )
+    (
+      {
+        cfg,
+        colors,
+        inputs,
+      }:
+      {
+        programs.${name}.profiles = lib.genAttrs cfg.profileNames (
+          _:
+          lib.mkIf cfg.firefoxGnomeTheme.enable {
```

Is there an issue with having the `mkIf` wrap the `genAttrs`?


```suggestion
        programs.${name}.profiles = lib.mkIf cfg.firefoxGnomeTheme.enable (
          lib.genAttrs cfg.profileNames (_: {
```